scripts/solitaire.js

```diff
@@ -0,0 +1,797 @@
+			name: 'queen',
+			colour: 'hearts',
+			weight: 12
+		},
+		{
+			name: 'king',
+			colour: 'hearts',
+			weight: 13
+		},
+		{
+			name: 'ace',
+			colour: 'hearts',
+			weight: 1
+		},
+		{
+			name: '2',
+			colour: 'clubs',
+			weight: 2
+		},
+		{
+			name: '3',
+			colour: 'clubs',
+			weight: 3
+		},
+		{
+			name: '4',
+			colour: 'clubs',
+			weight: 4
+		},
+		{
+			name: '5',
+			colour: 'clubs',
+			weight: 5
+		},
+		{
+			name: '6',
+			colour: 'clubs',
+			weight: 6
+		},
+		{
+			name: '7',
+			colour: 'clubs',
+			weight: 7
+		},
+		{
+			name: '8',
+			colour: 'clubs',
+			weight: 8
+		},
+		{
+			name: '9',
+			colour: 'clubs',
+			weight: 9
+		},
+		{
+			name: '10',
+			colour: 'clubs',
+			weight: 10
+		},
+		{
+			name: 'jack',
+			colour: 'clubs',
+			weight: 11
+		},
+		{
+			name: 'queen',
+			colour: 'clubs',
+			weight: 12
+		},
+		{
+			name: 'king',
+			colour: 'clubs',
+			weight: 13
+		},
+		{
+			name: 'ace',
+			colour: 'clubs',
+			weight: 1
+		},
+		{
+			name: '2',
+			colour: 'diamonds',
+			weight: 2
+		},
+		{
+			name: '3',
+			colour: 'diamonds',
+			weight: 3
+		},
+		{
+			name: '4',
+			colour: 'diamonds',
+			weight: 4
+		},
+		{
+			name: '5',
+			colour: 'diamonds',
+			weight: 5
+		},
+		{
+			name: '6',
+			colour: 'diamonds',
+			weight: 6
+		},
+		{
+			name: '7',
+			colour: 'diamonds',
+			weight: 7
+		},
+		{
+			name: '8',
+			colour: 'diamonds',
+			weight: 8
+		},
+		{
+			name: '9',
+			colour: 'diamonds',
+			weight: 9
+		},
+		{
+			name: '10',
+			colour: 'diamonds',
+			weight: 10
+		},
+		{
+			name: 'jack',
+			colour: 'diamonds',
+			weight: 11
+		},
+		{
+			name: 'queen',
+			colour: 'diamonds',
+			weight: 12
+		},
+		{
+			name: 'king',
+			colour: 'diamonds',
+			weight: 13
+		},
+		{
+			name: 'ace',
+			colour: 'diamonds',
+			weight: 1
+		}
+	]
+
+	var dropfields = [];
+
+	var mouseLeftDown = false;
+	var mouseDownCordsX = 0;
+	var mouseDownCordsY = 0;
+	var dragCardsElems = [];
+	var dragCardsElemsCoords = [];
+	var gameEnd;
+	var closedDeck;
+	var openedDeck;
+
+	var container = document.querySelector('[data-component="solitaireContainer"]');
+
+	var newGameBtn = container.querySelector('[data-component="newGameBtn"]');
+
+	initPage();
+
+	function initPage() {
+		
+		createDropfields();
+		createCardElems();
+		
+		container.addEventListener('mousedown', mouseDownTrue);
+		container.addEventListener('mousedown', flipCardOnClosedDeck);
+		container.addEventListener('mousedown', openCardOnClick);
+		container.addEventListener('contextmenu', autoCollectDecks);
+		container.addEventListener('mouseup', mouseDownFalse);
+		container.addEventListener('mousemove', dragCard);
+		newGameBtn.addEventListener('click',startNewGame);
+		
+		startNewGame();
+		
+	}
+	
+	function startNewGame() {
+		
+		gameEnd = false;
+		randomizeCards();
+		
+	}
+	
+	function autoCollectDecks(e) {
+		
+		if (gameEnd) {
+			return;
+		}
+		
+		e.preventDefault();
+		
+		var finalDropdileds = [];
```

Comment:
move logic code to it's own function from event handler.
otherwise you'll be forced to mock more data while unit testing